src/org/javarosa/core/services/ProgramFlow.java

```diff
@@ -0,0 +1,32 @@
+package org.javarosa.core.services;
+
+import org.javarosa.core.log.FatalException;
+import org.slf4j.LoggerFactory;
+
+public final class ProgramFlow {
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(ProgramFlow.class);
+
+    public static void die(String thread, Exception e) {
```

Comment:
Maybe add a doc comment, explaining what this is all about—why it’s needed.

Reply:
This is actually old code that I've only moved around... I couldn't know. 

I mean, I can interpret that there's some intent of ending "gracefully" normal execution flow, but not much more...

Reply:
Sorry, I thought it was new.